workspace_defs.bzl

```diff
@@ -47,221 +31,80 @@ def google_common_workspace_rules():
-    )
-
-    _maven_import(
-        artifact = "net.bytebuddy:byte-buddy-agent:1.9.10",
-        licenses = ["notice"],
-        sha256 = "8ed739d29132103250d307d2e8e3c95f07588ef0543ab11d2881d00768a5e182",
-    )
-
-    _maven_import(
-        artifact = "org.mockito:mockito-core:2.28.2",
-        licenses = ["notice"],
-        sha256 = "b0af36fed3a6c2147c0cd9028a1d814fd4f4e8196c539f2befddb61ca6ec9e27",
-    )
-
-    _maven_import(
-        artifact = "org.hamcrest:hamcrest-core:1.3",
-        licenses = ["notice"],
-        sha256 = "66fdef91e9739348df7a096aa384a5685f4e875584cce89386a7a47251c4d8e9",
-    )
-
-    _maven_import(
-        artifact = "org.objenesis:objenesis:1.0",
-        licenses = ["notice"],
-        sha256 = "c5694b55d92527479382f254199b3c6b1d8780f652ad61e9ca59919887f491a8",
-    )
-
-    _maven_import(
-        artifact = "com.google.truth:truth:0.45",
-        licenses = ["notice"],
-        sha256 = "0f7dced2a16e55a77e44fc3ff9c5be98d4bf4bb30abc18d78ffd735df950a69f",
-    )
-
-    _maven_import(
-        artifact = "com.google.truth.extensions:truth-java8-extension:0.45",
-        licenses = ["notice"],
-        sha256 = "dc1fedf6c13b1b1a4c4fa3e810f11df070ea4701765f05176f8bdcb5520c7de4",
-    )
-
-    _maven_import(
-        artifact = "com.squareup:javapoet:1.11.1",
-        licenses = ["notice"],
-        sha256 = "9cbf2107be499ec6e95afd36b58e3ca122a24166cdd375732e51267d64058e90",
-    )
-
-    _maven_import(
-        artifact = "io.grpc:grpc-core:1.2.0",
-        licenses = ["notice"],
-        sha256 = "4434ffd957dc5ca752d8a8e6e71fa6d598a05bb02b4fc08e48e53d878a004ee5",
-    )
-
-    _maven_import(
-        artifact = "io.grpc:grpc-netty:1.2.0",
-        licenses = ["notice"],
-        sha256 = "c9379d17fdec2eae203679495a695b523e01f2541169d28f5b780de298aa17c8",
-    )
-
-    _maven_import(
-        artifact = "io.grpc:grpc-context:1.2.0",
-        licenses = ["notice"],
-        sha256 = "4f1fed2735f011ba6f8ab1faa003ef67bade9e773f5a5ec4b69eb2a124500ca6",
-    )
-
-    _maven_import(
-        artifact = "io.grpc:grpc-protobuf:1.2.0",
-        licenses = ["notice"],
-        sha256 = "19797fc26192dfcc4570ec26c12ba84583842b0ccbcd7d54982f922d33209383",
-    )
-
-    _maven_import(
-        artifact = "io.grpc:grpc-stub:1.2.0",
-        licenses = ["notice"],
-        sha256 = "bf3eae95175ed36eee086d5fb320583fc492b144bd733d6e19515c7568ee2e2b",
-    )
-
-    _maven_import(
-        artifact = "io.grpc:grpc-all:1.2.0",
-        licenses = ["notice"],
-        sha256 = "6b697a05b203216b853394d276c429da243cdf50f519688b33f4edbbf5f126d7",
-    )
+    GRPC_VERSION = "1.2.0"
+    LOG4J2_VERSION = "2.11.2"
+    INCAP_VERSION = "0.2"
 
-    _maven_import(
-        artifact = "com.google.protobuf:protobuf-java:3.7.0",
-        licenses = ["notice"],
-        sha256 = "dc7f93e3a3dc2c11be5ba9672af3e26410f0a3289312dbf2260d4d8a0c711a51",
+    maven_install(
+        artifacts = [
```

Comment:
I recommend making this list available as a variable for downstream consumers to import into their local maven_install instance. See https://github.com/bazelbuild/rules_jvm_external/#exporting-and-consuming-artifacts-from-external-repositories

Reply:
Does that mean that we should only call `maven_install` in the `WORKSPACE` and rely on every user to call `maven_install` with both their dependencies and this list?

Reply:
Yeah, this has the advantage of ensuring that's only one version of every artifact in a single namespace (`@maven`). I'd think that most bazel-common projects would also need to pull other maven artifacts as well.

Do you think this is a good user experience?

Reply:
While I definitely see the value into having a single version of the dependencies, this requires downstream consumer to have a call to `maven_install`, while the purpose of bazel-common seems to be to minimise the set-up required in downstream workspaces. Maybe @ronshapiro would have an input on that?

It seems that we have two alternatives: either downstream consumers have to call `maven_install` and pass in bazel-common's list of artifacts, or we can call `maven_install` in bazel-common's set-up with a specific repository name, downstream consumers don't need to do anything, but if they use some Maven dependencies not provided by bazel-common, they're likely to end up with different versions of transitive dependencies that could have been provided by bazel-common (which is the current behaviour, as far as I can tell).

Reply:
Yeah, I'd prefer the former, but the latter works as well. I'll leave that up to the maintainers of this repo to decide.